### ISSUE TITLE
Failed finding central directory

### ISSUE THREAD

**Issue by slush0** [bigscience-workshop/petals#236]
I believe the server got stuck on following error, because this is the last output in console and it is many hours old. There's no following line about reading the block from HF. Restarting the server helped and it works normally now.

It might be cause by network downtime, but I suppose the server should survive short outages like this.

```
Jan 24 06:36:00.449 [WARN] [/home/petals/src/petals/bloom/from_pretrained.py._load_state_dict:121] Failed to load block 64 from HF Hub (retry in 10240 sec)
Traceback (most recent call last):
  File "/home/petals/src/petals/bloom/from_pretrained.py", line 118, in _load_state_dict
    return torch.load(archive_file, map_location="cpu")
  File "/opt/conda/lib/python3.10/site-packages/torch/serialization.py", line 777, in load
    with _open_zipfile_reader(opened_file) as opened_zipfile:
  File "/opt/conda/lib/python3.10/site-packages/torch/serialization.py", line 282, in __init__
    super(_open_zipfile_reader, self).__init__(torch._C.PyTorchFileReader(name_or_buffer))
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory
```

**Comment by slush0** [comment]
Is possible that the file got corrupted or partially downloaded and the reader didn't recover from that?

**Comment by borzunov** [comment]
@slush0 Apparently, that's indeed what happened. The simplest solution would be to run `rm ~/.cache/petals` and let it recreate the disk cache from scratch.

If you want to remove only the corrupted block (block64) instead of redownloading all blocks, you can use [delete_revisions() method](https://huggingface.co/docs/huggingface_hub/v0.11.0/en/how-to-cache#clean-cache-from-python) from the `huggingface_hub` library that we use for caching HF models. I can write a command for you, let me know if you need this.

**Comment by slush0** [comment]
As I have downloaded ~200GB of blocks already, I would definitely prefer helper for deleting specific blocks. That could be also part of docs as I believe this can happen to other people eventually, too.

**Comment by slush0** [comment]
Ok, I find out tool ```huggingface-cli delete-cache --dir .``` and the UI is pretty straightforward, but I realized that the TUI does not show block 64. The listing ends with "block 62". Is that possible that actually the tmp folder got corrupted and it cannot successfully continue the download? I'm not familiar how the orchestration in hub works.

I deleted all temp files in hub and retry the server...
